ui/src/main/java/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -8945,36 +8921,6 @@ public Repository getRepository() {
   public void setRepository( Repository rep ) {
     this.rep = rep;
     this.repositoryName = rep != null ? rep.getName() : null;
-    try {
-
-      // Keep one metastore here...
-      //
-      if ( metaStore.getMetaStoreList().size() > 1 ) {
-        metaStore.getMetaStoreList().remove( 0 );
-        metaStore.setActiveMetaStoreName( metaStore.getMetaStoreList().get( 0 ).getName() );
-      }
-
-      if ( rep != null ) {
```

Comment:
I am so sorry for adding comments in bit and pieces but now I understand the code little more and I feel like the line 8957 -8958 should stay as it is not related to metastore
My understanding is as User is switching from  one repository to another, the RepositoryCapabilities should be set to current repository. With this change, our RepositoryCapabilities won't be set to the newly connected repository.

Reply:
Good catch.